### PR TITLE
[tune/train] clean up tune/train result output

### LINKED ISSUE #32111: [tune/train] remove train's auto-filled fields like `_training_iteration` etc that are duplicated with tune
### What happened + What you expected to happen

train session doesn't keep track of `_training_iteration` correctly - it's always resetting to 0 upon restore. 
tune session already has another one called `training_iteration`. 

We should remove those duplicated ones as train is always executed using tune now.

The same for `_time_this_iter_s`.

### Versions / Dependencies

master

### Reproduction script

NA

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by xwjiang2010**
Signed-off-by: xwjiang2010 <xwjiang2010@gmail.com>

Clean up some results reported by tune/train.
* remove duplicated keys in tune/train results.
* `timestamp` and `time_this_iter_s` from train should triumph tune.
* remove `warmup_time`, which is not very useful and not mentioned anywhere in our doc.
* remove some of the fields in tune that are overfitting to rllib's case.

## Related issue number
Closes #32111

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/test_utils.py [hunk 1]
```diff
@@ -634,7 +634,6 @@ def check_train_results(train_results: PartialAlgorithmConfigDict) -> ResultDict
         "episode_reward_max",
         "episode_reward_mean",
         "episode_reward_min",
-        "episodes_total",
```
Comment (xwjiang2010) [inline rllib/utils/test_utils.py:637]: These two are not populated by rllib anymore. 
We probably should just remove them from Tune as well..

File: python/ray/train/constants.py [hunk 2]
```diff
@@ -16,21 +16,20 @@
...
+TIME_THIS_ITER_S = "time_this_iter_s"
 
-BASIC_AUTOFILLED_KEYS = {TIMESTAMP, TIME_THIS_ITER_S, TRAINING_ITERATION}
-
-DATE = "_date"
+TIME_TOTAL_S = "_time_total_s"
+# Note the following 3 are semantically different than what's auto-filled by Tune.
 HOSTNAME = "_hostname"
 NODE_IP = "_node_ip"
 PID = "_pid"
```
Comment (krfricke) [inline python/ray/train/constants.py:28]: Can we rename these constants to `WORKER_HOSTNAME`, `WORKER_NODE_IP` etc do make this clear?

File: python/ray/train/constants.py [hunk 3]
```diff
@@ -16,21 +16,20 @@
 )
 
 # Autofilled session.report() metrics. Keys should be consistent with Tune.
-TIMESTAMP = "_timestamp"
-TIME_THIS_ITER_S = "_time_this_iter_s"
-TRAINING_ITERATION = "_training_iteration"
+# The train provided `TIME_THIS_ITER_S` and `TIMESTAMP` will triumph what's
+# auto-filled by Tune session.
+TIMESTAMP = "timestamp"
+TIME_THIS_ITER_S = "time_this_iter_s"
```
Comment (krfricke) [inline python/ray/train/constants.py:22]: If they need to be consistent, can we just import them from tune?

**Review by krfricke — APPROVED** [review]
Generally LGTM!

File: python/ray/train/constants.py (hunk 3 above)
Comment (xwjiang2010) [reply python/ray/train/constants.py:22]: yeah makes sense, but can we do that after this? https://github.com/ray-project/ray/issues/32260

Comment (xwjiang2010) [reply python/ray/train/constants.py:24]: Also added a TODO

**Pull request merged**